include/aws/s3/private/s3_auto_ranged_put.h

```diff
@@ -25,6 +25,8 @@ struct aws_s3_auto_ranged_put {
     /* Initialized either during creation in resume flow or as result of create multipart upload during normal flow. */
     struct aws_string *upload_id;
 
+    struct aws_s3_meta_request_resume_token *resume_token;
```

Comment:
added comment. the only reason its not const is because we call release/acquire on it and as mentioned in another comment those require non-const unfortunately.